### ISSUE TITLE
Add user behavior clustering (K-means segmentation)

### ISSUE THREAD

**Issue by evanvolgas** [ashita-ai/conduit#49]
## Summary

Cluster users into behavioral segments (cost_optimizers, speed_seekers, quality_focused, balanced_users) using K-means on behavior metrics.

## Problem

Explicit preferences (#47) tell us what users WANT. But behavior shows what they ACTUALLY value. Clustering discovers natural user segments based on observed behavior.

## Solution: K-means Clustering (Option 4)

Use scikit-learn to cluster users into 4 segments based on behavior metrics.

```python
from sklearn.cluster import KMeans

class UserSegmenter:
    SEGMENTS = {
        0: "cost_optimizers",
        1: "speed_seekers",
        2: "quality_focused",
        3: "balanced_users"
    }
    
    async def fit(self, behaviors: list[UserBehavior]):
        '''Train clustering model.'''
        X = np.array([[
            b.avg_cost_per_query,
            b.avg_latency_seconds,
            b.avg_quality_score,
            b.cost_sensitivity,
            b.latency_sensitivity,
            b.quality_sensitivity
        ] for b in behaviors])
        
        scaler = StandardScaler()
        X_scaled = scaler.fit_transform(X)
        self.clusterer.fit(X_scaled)
    
    async def predict_segment(self, behavior: UserBehavior) -> str:
        '''Assign user to segment.'''
        X = [...]  # same features
        cluster_id = self.clusterer.predict(X)[0]
        return self.SEGMENTS[cluster_id]
```

## Database Schema

```sql
CREATE TABLE user_segments (
    user_id TEXT PRIMARY KEY,
    segment_id TEXT NOT NULL,
    confidence FLOAT NOT NULL,
    last_updated TIMESTAMP NOT NULL,
    total_queries INT NOT NULL,
    
    -- Behavior metrics (for analysis)
    avg_cost_per_query FLOAT,
    avg_latency_seconds FLOAT,
    avg_quality_score FLOAT,
    cost_sensitivity FLOAT,
    latency_sensitivity FLOAT,
    quality_sensitivity FLOAT
);

-- Segment-specific bandit state
ALTER TABLE model_states ADD COLUMN segment_id TEXT;
CREATE INDEX idx_model_states_segment ON model_states(segment_id, model_id);
```

## Implementation Tasks

- [ ] Create `conduit/ml/segmenter.py` module
- [ ] Implement `UserSegmenter` class with K-means
- [ ] Add `user_segments` table migration
- [ ] Add `segment_id` column to `model_states` table
- [ ] Implement weekly re-clustering background job
- [ ] Add segment assignment to database service
- [ ] Add silhouette score validation
- [ ] Add tests for clustering
- [ ] Add example: `examples/05_personalization/behavior_clustering.py`

## Expected Impact

- **Cold start**: Uses explicit pref (#47) until 10+ queries
- **Learning**: Discovers natural user segments from data
- **Accuracy**: Behavior > stated preferences (revealed preferences)
- **Multi-tenancy filtering**: Critical for segment-based routing
- **Effort**: 2-3 days implementation

## Related

- Depends on: #48 (Behavior tracking)
- Works with: #47 (Explicit preferences - cold start fallback)
- Enables: #49 (Hybrid routing)
- Parent: #29 (User-level personalization)
